### PR TITLE
add opts_for_module/2 and return opts on stop_module

### PR DESCRIPTION
**Pull request by sstrigler**
Admin love: 
- adds a function gen_mod:opts_for_module/2 which allows you to selectively retrieve opts passed through config to a given module
- changes stop_module_keep_config/2 and stop_module/2 to return those opts

### CODE REVIEW

**Comment by michalwski** [comment]
Hi @sstrigler 

Thanks a lot for this PR. Proposed changes make sense.
I have only one problem with it. All jobs on travis-ci failed, this is probably not a random failure :)

**Comment by michalwski** [comment]
@sstrigler any chance you'll have a look at the failed tests?

File: test/ejabberd_tests/tests/login_SUITE.erl [hunk 1]
```diff
@@ -613,7 +613,7 @@ restart_mod_register_with_option(Config, Name, Value) ->
     Domain = escalus_config:get_config(ejabberd_domain, Config),
     ModuleOptions = escalus_ejabberd:rpc(gen_mod, loaded_modules_with_opts, [Domain]),
```
Comment (gadgetci) [inline test/ejabberd_tests/tests/login_SUITE.erl]: According to **Elvis**:
> Line 614 is too long:     ModuleOptions = escalus_ejabberd:rpc(gen_mod, loaded_modules_with_opts, [Domain]),.

File: test/ejabberd_tests/tests/login_SUITE.erl [hunk 2]
```diff
@@ -613,7 +613,7 @@ restart_mod_register_with_option(Config, Name, Value) ->
...
     {mod_register, OldRegisterOptions} = lists:keyfind(mod_register, 1, ModuleOptions),
```
Comment (gadgetci) [inline test/ejabberd_tests/tests/login_SUITE.erl]: According to **Elvis**:
> Line 615 is too long:     {mod_register, OldRegisterOptions} = lists:keyfind(mod_register, 1, ModuleOptions),.

**Comment by fenek** [comment]
Hi @sstrigler 

Is there a chance that you could rebase and finalise this PR?

**Comment by sstrigler** [comment]
Ok, will do. :)

**Comment by sstrigler** [comment]
I don't get this codecov stuff

**Comment by michalwski** [comment]
@sstrigler thanks for the updates to this PR. Regarding the code coverage:
* coverage/coveralls status can be ignored - this site sometimes looses some reports that's why it may happen it reports lower coverage than in fact is :(
* codecov/patch - thats the coverage of your diff only. Looks like the 55.55% is not correct looking at the covered lines.

Anyways, all looks good to me, I'm merging.

**Pull request merged**